README.md

```diff
@@ -8,22 +8,845 @@
+  ) {
+    // Configures a permissioned deploy protection as well as a cross-chain redeploy protection.
+    guardedSalt = keccak256(abi.encode(msg.sender, block.chainid, salt));
+  } else if (
+    senderBytes == SenderBytes.MsgSender &&
+    redeployProtectionFlag == RedeployProtectionFlag.False
+  ) {
+    // Configures solely a permissioned deploy protection.
+    guardedSalt = _efficientHash({
+      a: bytes32(uint256(uint160(msg.sender))),
+      b: salt
+    });
+  } else if (senderBytes == SenderBytes.MsgSender) {
+    // Reverts if the 21st byte is greater than `0x01` in order to enforce developer explicitness.
+    revert InvalidSalt({ emitter: _SELF });
+  } else if (
+    senderBytes == SenderBytes.ZeroAddress &&
+    redeployProtectionFlag == RedeployProtectionFlag.True
+  ) {
+    // Configures solely a cross-chain redeploy protection. In order to prevent a pseudo-randomly
+    // generated cross-chain redeploy protection, we enforce the zero address check for the first 20 bytes.
+    guardedSalt = _efficientHash({ a: bytes32(block.chainid), b: salt });
+  } else if (
+    senderBytes == SenderBytes.ZeroAddress &&
+    redeployProtectionFlag == RedeployProtectionFlag.Unspecified
+  ) {
+    // Reverts if the 21st byte is greater than `0x01` in order to enforce developer explicitness.
+    revert InvalidSalt({ emitter: _SELF });
+  } else {
+    // In all other cases, the salt value `salt` is not modified.
+    guardedSalt = salt;
+  }
+}
+```
+
+Please note that when you configure a permissioned deploy protection, you **must** specify whether you want a cross-chain redeploy protection or not (i.e. the 21st byte equals `0x00`). The underlying reason for this logic is to enforce developer explicitness. If you don't configure a cross-chain redeploy protection, i.e. the 21st byte is greater than `0x01`, the function reverts. Furthermore, you can configure solely a cross-chain redeploy protection by setting the first 20 bytes equal to the zero address `0x0000000000000000000000000000000000000000`. The rationale behind this logic is to prevent a pseudo-randomly generated cross-chain redeploy protection. Also in this case, if you don't specify a cross-chain redeploy protection, i.e. the 21st byte is greater than `0x01`, the function reverts. The underlying reason for this logic is as well to enforce developer explicitness.
+
+### Pseudo-Random Salt Value
+
+For developer convenience, the [`CreateX`](./src/CreateX.sol) contract offers several overloaded functions that generate the salt value pseudo-randomly using a diverse selection of block and transaction properties. Please note that this approach does not guarantee true randomness!
+
+The full logic is implemented in the `internal` [`_generateSalt`](./src/CreateX.sol#L965-L991) function:
+
+```solidity
+function _generateSalt() internal view returns (bytes32 salt) {
+  // If you use this function between the genesis block and block number 31, it will return zero.
```

Comment:
Ah this is still a bit misleading actually. In that block range, the value of `blockhash(block.number - 32)` is zero, but the resulting salt from the entire hash is still some nonzero value based on the remaining properties

Reply:
Great callout! Adjusted via https://github.com/pcaversaccio/createx/commit/1dcdf248121d2d503b5ac4aad4db08c1a77b3a03.